### ISSUE TITLE
Mimic std/httpclient exception behaviour 

### ISSUE THREAD

**Issue by ire4ever1190** [treeform/puppy#31]
Puppy currently just returns an empty string if a error occurs when using the basic `fetch(string)` proc. Could there be a way to mimic the [behaviour](https://github.com/nim-lang/Nim/blob/version-1-4/lib/pure/httpclient.nim#L1079) of std/httpclient in that an exception is thrown for any 4xx or 5xx http code, maybe through a compile switch or proc argument so that current code doesn't break?

**Comment by guzba** [comment]
I do not agree with 4xx or 5xx http status codes resulting in exceptions. That seems very strange to me. We'll see if treeform wants that but I do not support it.

If you want that behavior, a small wrapper could reproduce that behavior in your project without much trouble.

**Comment by dom96** [comment]
Even in little scripts I would say that getting an exception would be useful. Throwing on non-200 doesn't make this much more complex.

But if we're designing an API for scripts and another for non-scripts, maybe both should be in separate modules?
